packages/slice-machine/src/domain/CustomTypeModel.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  CustomType,
+  DynamicSection,
+  DynamicSlicesConfig,
+} from "@prismicio/types-internal/lib/customtypes";
+
+export const CustomTypeModel = {
```

Comment:
Hello @hypervillain, @lihbr, @angeloashmore

I discussed a bit with Baptiste on two things:

- Having shared functions in `src/domains` like Baptiste did in `src/domains/slice.ts` and I did on this file. We know there is existing shared functions, but we feel either some are missing or too complicated. We would also like to push for not relying on specific SM types, but only the ones shared with the managers.
- We also discussed how to share this kind of shared domain functions:
  - a file like `src/domains/slice.ts` where functions are directly exported
  - a file like `src/domains/CustomTypeModel.ts` where an object that contains all the functions is exported
  
It's mostly for the second part that we would like to have your preferences. We discussed together a bit the two, and we would like to see what you prefer before giving you our arguments 🙂 
No hard blocker for one or the other, just good to agree on a way to do things.

cc @bapmrl

Reply:
I prefer option 1 imo `src/domains/slices.ts`

Reply:
I have a small preference for objects, I don't like the micro frustrations of forgetting the name of an import, or an import not being completed properly. We don't have large perf problems so let's use this fact to be simpler!

Reply:
Baptiste also proposed we you can do with option one:

`import * as CustomTypeModel from "src/features/custom-type.ts"`

If you don't know the function name exactly, for example.

Reply:
Done, the `import * as CustomTypeModel from "src/features/custom-type.ts"` is done as an example in the test file. Less imports so.